src/main/java/com/algorand/algosdk/builder/transaction/ApplicationCallReferencesSetter.java

```diff
@@ -0,0 +1,30 @@
+package com.algorand.algosdk.builder.transaction;
+
+import java.util.List;
+
+import com.algorand.algosdk.crypto.Address;
+
+public interface ApplicationCallReferencesSetter<T extends ApplicationCallReferencesSetter<T>> {
```

Comment:
thanks for the discussion https://github.com/algorand/java-algorand-sdk/pull/324#discussion_r882194581, and now I understand the reason behind the syntax `T extends ...<T>` among all the txn builder.